qa/rpc-tests/feature_zip244_blockcommitments.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Zcash developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+
+from test_framework.blocktools import derive_block_commitments_hash
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    bytes_to_hex_str,
+    hex_str_to_bytes,
+    start_nodes,
+)
+
+TERMINATOR = b'\x00' * 32
+
+# Verify block header field 'hashLightClientRoot' is set correctly for NU5 blocks.
+class AuthDataRootTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 4
+
+    def setup_nodes(self):
+        return start_nodes(self.num_nodes, self.options.tmpdir, extra_args=[[
+            '-nuparams=5ba81b19:1', # Overwinter
+            '-nuparams=76b809bb:1', # Sapling
+            '-nuparams=2bb40e60:201', # Blossom
+            '-nuparams=f5b9230b:201', # Heartwood
+            '-nuparams=e9ff75a6:201', # Canopy
+            '-nuparams=f919a198:205', # NU5
+            '-nurejectoldversions=false',
+        ]] * self.num_nodes)
+
+    def run_test(self):
+        # Generate a block so we are on Canopy rules.
+        self.nodes[0].generate(2)
+        self.sync_all()
+
+        # Before NU5 activation, the hashBlockCommitments field of the block
+        # header contains the root of the ZIP 221 history tree.
+        block_before = self.nodes[0].getblock('202')
+        assert_equal(block_before['blockcommitments'], block_before['chainhistoryroot'])
+
+        # Generate blocks until we are on NU5 rules.
+        self.nodes[0].generate(3)
```

Comment:
Why do we need to generate three blocks here? Shouldn't we be testing the block before, the activation block, and the block after?

Reply:
It's 3 because I'm using the cached chain which is active up to Sapling, so needed to go to 202 to get a block with any chain history root in it, and then I'd picked 205 for NU5 activation arbitrarily. But yes, I can adjust this to run the exact same check on the intervening blocks as well as the block after.

Reply:
My plan is that once we have support for creating v5 transactions, we will amend this test to check that:
- Mining a block without v5 transactions alters the root in the expected way (via predictably-placed empty or legacy hashes).
- Mining a block with a v5 transaction alters the root in the expected way (we probably won't check the exact value in-line, but it should not be the same as the no-v5 case).